wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -187,7 +227,152 @@ export default {
           }
         }
       })(_this);
-    }
+    },
+
+    process: function(data, query, headers) {
+      // If there are posts for this query, map them to the template.
+      let posts = (Array.isArray(data)) ?
+        data.map(this.maps()[this.type]) : false;
+
+      // Set posts and store a copy of the query for reference.
+      this.$set(this.posts[query.page], 'posts', posts);
+      this.$set(this.posts[query.page], 'headers', Object.freeze(headers));
+
+      // If there are no posts, pass along to the error handler.
+      if (!Array.isArray(data))
+        this.error({error: data, query: query});
+
+      this.$set(this, 'init', true);
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+        this.totalPages = [2,3,4,5]
+      } 
+    },
+
+    updatePagination: function(){
+      this.scrollToTop();
+      let url = [
+        this.domain,
+        this.lang.path,
+        this.endpoints[this.type],
+        this.buildUrlQuery(this.query)
+      ].join('');
+
+      fetch(url)
+      .then(this.response)
+      .then(data => {
+      let query = this.query;
+      let headers = Object.assign({}, this.headers);
+
+      this.posts.length=1;
+
+      let currentPost = {
+        'posts': data,
+        'headers': headers,
+        'query': this.query,
+        'show': true
+      };
+
+      this.posts.push(currentPost);
+
+      this.posts.map(post=>{
+        if(post.query.page != query.page){
+          this.$set(post, 'show', false);
+        }
+      });
+      this.setPagination();      
+      }).catch(this.error);
+    },
+
+    previousPage:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage-1);
+      this.updatePagination();
+    },
+
+    nextPagination:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage+1);
+      this.updatePagination();
+    },
+
+    immediatePage: function(event){
+      let change = parseInt(event.target.dataset.amount);
+      this.$set(this.query, 'page', change);
+      this.updatePagination();
+    },
+
+    updateQuery: function(taxonomy, terms) {
+      return new Promise((resolve) => { // eslint-disable-line no-undef
+        this.$set(this.query, taxonomy, terms);
+        this.$set(this.query, 'page', 1);
+        // hide all of the posts
+        this.posts.map((value, index) => {
+          //if (value) this.$set(this.posts[index], 'show', false);
+          if (value) this.posts.length=1;
+          return value;
+        });
+        resolve();
+      })
+      .then(this.wp)
+      .catch(message => {
+      });
+    },
+
+    setPagination: function(){
+      if(this.headers.pages<=6){
```

Comment:
The logic in `setPagination` is relatively complex, so can you add comments explaining the steps within the function?